src/irmin-pack/checks_intf.ml

```diff
@@ -92,7 +89,17 @@ module type Checks = sig
 
   module type Subcommand = Subcommand
   module type S = S
-  module type Make_args = Make_args
 
-  module Make (_ : Make_args) : S
+  module Make
+      (Conf : Config.S)
+      (M : Irmin.Metadata.S)
+      (C : Irmin.Contents.S)
+      (P : Irmin.Path.S)
+      (B : Irmin.Branch.S)
+      (H : Irmin.Hash.S)
+      (Node : Irmin.Private.Node.S
+                with type metadata = M.t
+                 and type hash = H.t
+                 and type step = P.step)
+      (Commit : Irmin.Private.Commit.S with type hash = H.t) : S
```

Comment:
I think it'd be nice to avoid exposing all of this intermediate state if possible. Is this just to support being able to construct stores at different versions? If so, it might be possible to rename `Make_args.Store` to `Make_args.Versioned_store` and make it a functor from `Io_version` instead?

Reply:
New signature for `Check.Make`, let me know what you think, and I'll squash the commits. 

Also we don't need the layered/ext.ml anymore, but I think we can keep as it mirrors nicely with the file in irmin-pack (but I can remove too).

Reply:
Looks good to me, thanks!

Reply:
Squashed commits and rebased.